test/control.js

```diff
@@ -79,3 +79,58 @@ metatests.test('sequential with error', test => {
     test.end();
   });
 });
+
+metatests.test('runIf run asyncFn', test => {
+  const asyncFn = test.mustCall((arg1, arg2, cb) => {
+    process.nextTick(() => {
+      cb(null, { arg1, arg2 });
+    });
+  });
+
+  metasync.runIf(true, asyncFn, 'val1', 'val2', (err, result) => {
+    test.error(err);
+    test.strictSame(result, { arg1: 'val1', arg2: 'val2' });
+    test.end();
+  });
+});
+
+metatests.test('runIf do not run asyncFn', test => {
+  const asyncFn = test.mustNotCall((arg1, arg2, cb) => {
```

Comment:
And same here. `const asyncFn = test.mustNotCall()`, wdyt?

Reply:
This will not work, because `asyncFn` must call callback inside.

Reply:
You mean it doesn't have a proper signature of an async function? Well, it looks ok to me though if you think it'll be clearer with verbose version feel free to leave it as is.